packages/eslint-plugin/tests/rules/no-floating-promises.test.ts

```diff
@@ -656,6 +656,68 @@ async function test() {
         {
           line: 3,
           messageId: 'floating',
+          suggestions: [
+            {
+              messageId: 'floatingFixAwait',
+              output: `
+async function test() {
+  await Promise.resolve();
+}
+      `,
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+async function test() {
+  const promise = new Promise((resolve, reject) => resolve('value'));
+  promise;
+}
+      `,
+      options: [{ ignoreVoid: false }],
+      errors: [
+        {
+          line: 4,
+          messageId: 'floating',
+          suggestions: [
+            {
+              messageId: 'floatingFixAwait',
+              output: `
+async function test() {
+  const promise = new Promise((resolve, reject) => resolve('value'));
+  await promise;
+}
+      `,
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+async function returnsPromise() {
+  return 'value';
+}
+void returnsPromise();
```

Comment:
One more thing to test here:

> A variation or two of this that uses `/* ... */` comments or similar after the void - to make sure text ranges are accounted for well

I just want to make sure that `void /* aha */ returnsPromise()` gets converted to something reasonable like `await /* aha */ returnsPromise()`.

Reply:
@JoshuaKGoldberg 
Thank you!
I understand.

I've added test case.
1ddb23b5f3e731808b3928836d1db30ad9965c5d